include/seqan3/alignment/matrix/alignment_trace_algorithms.hpp

```diff
@@ -78,30 +79,26 @@ inline alignment_coordinate alignment_front_coordinate(trace_matrix_t && matrix,
 
 /*!\brief Compute the trace from a trace matrix.
  * \ingroup alignment_matrix
+ * \tparam    alignment_t                The type of the returned alignment.
  * \tparam    database_t                 The type of the database sequence.
  * \tparam    query_t                    The type of the query sequence.
  * \tparam    trace_matrix_t             The type of the trace matrix.
- * \cond DEV
- * \tparam    gapped_database_alphabet_t The alphabet type of the gapped database sequence.
- * \tparam    gapped_query_alphabet_t    The alphabet type of the gapped query sequence.
- * \endcond
  * \param[in] database                   The database sequence.
  * \param[in] query                      The query sequence.
  * \param[in] matrix                     The trace matrix.
  * \param[in] back_coordinate            Where the trace in the matrix ends.
  * \returns Returns a seqan3::aligned_sequence.
  */
 template <
+    typename alignment_t,
     typename database_t,
```

Comment:
What are the constraints for this?

Reply:
I have just discussed with @rrahn that for the release we will enforce that all ranges are RandomAccess and Sized. This means you don't need a workaround for copying non randomAccess ranges.